components/ActivityForm/index.js

```diff
@@ -98,14 +107,14 @@ export default function ActivityForm({ activity, onSubmit, isUpdateMode }) {
           defaultValue={activity?.area}
         ></StyledInputs>
         <label htmlFor="country">Country: </label>
-        <StyledInputs
+        <StyledSelect
```

Comment:
Your Select component doesn't knows the current country in 'update'-mode. But it renders it as a placeholder. That leads to the bug that the country gets lost when you update an activity without changing (or re-selecting) the country.